### PR TITLE
Update README.md with common structure

### PR DESCRIPTION
**Pull request by gogasca**
Update README.md with a common template.

---
This change is [Reviewable](https://reviewable.io/reviews/googlecloudplatform/cloudml-samples/310)

### CODE REVIEW

File: mnist/tensorflow/keras/fashion/README.md [hunk 1]
```diff
@@ -95,15 +100,16 @@ gsutil cp -r data/train-images-idx3-ubyte.gz gs://your-bucket-name/t10k-labels-i
 ```
 MNIST_DATA=data
 DATE=`date '+%Y%m%d_%H%M%S'`
-export JOB_DIR=mnist_$DATE
+export JOB_DIR=boston_$DATE
```
Comment (nnegrey) [inline mnist/tensorflow/keras/fashion/README.md]: Shouldn't this still be mnist?

File: mnist/tensorflow/keras/fashion/README.md [hunk 2]
```diff
@@ -114,21 +120,21 @@ python -m trainer.task \
...
 export JOB_NAME="mnist_keras_$(date +%Y%m%d_%H%M%S)"
 export JOB_DIR=gs://$BUCKET_NAME/$JOB_NAME
-export REGION=us-central1
 export TRAIN_FILE=gs://cloud-samples-data/ml-engine/mnist/train-images-idx3-ubyte.gz
 export TRAIN_LABELS_FILE=gs://cloud-samples-data/ml-engine/mnist/train-labels-idx1-ubyte.gz
 export TEST_FILE=gs://cloud-samples-data/ml-engine/mnist/t10k-images-idx3-ubyte.gz
 export TEST_LABELS_FILE=gs://cloud-samples-data/ml-engine/mnist/t10k-labels-idx1-ubyte.gz
 ```
 
+* **Run locally in Google Cloud ML Engine:**
```
Comment (nnegrey) [inline mnist/tensorflow/keras/fashion/README.md]: We should rename this, it just seems confusing. 

"Run locally via the gcloud emulator for Google Cloud ML Engine"?

File: census/README.md [hunk 3]
```diff
@@ -1,313 +1,324 @@
...
 before `--` while training on the cloud and this is so that we can have
 different trial runs during Hyperparameter tuning.
 
+* **GCloud configuration:**
+
 ```
 DATE=`date '+%Y%m%d_%H%M%S'`
 export JOB_NAME=census_$DATE
-export GCS_JOB_DIR=gs://<my-bucket>/path/to/my/jobs/$JOB_NAME
+export GCS_JOB_DIR=gs://your-bucket-name/to/my/jobs/$JOB_NAME
```
Comment (nnegrey) [inline census/README.md]: `path` might be worth keeping. Though I prefer we just suggest a real name to use.

File: mnist/tensorflow/keras/fashion/README.md (hunk 1 above)
Comment (gogasca) [reply mnist/tensorflow/keras/fashion/README.md]: ack

File: mnist/tensorflow/keras/fashion/README.md (hunk 2 above)
Comment (gogasca) [reply mnist/tensorflow/keras/fashion/README.md]: ack

File: census/README.md (hunk 3 above)
Comment (gogasca) [reply census/README.md]: ack

File: boston/README.md [hunk 4]
```diff
@@ -123,19 +122,18 @@ python -m trainer.task \
...
 DATE=`date '+%Y%m%d_%H%M%S'`
 export JOB_NAME="boston_keras_$DATE"
-export JOB_DIR=gs://$BUCKET_NAME/$JOB_NAME
-export REGION=us-central1
+export JOB_DIR=boston_$DATE
+rm -rf $JOB_DIR
 export TRAIN_FILE=gs://cloud-samples-data/ml-engine/boston/boston_housing.npz
 ```
-* **Run locally:**
+* **Run locally via the gcloud emulator for Google Cloud ML Engine:**
```
Comment (dizcology) [inline boston/README.md]: Not sure if it is really an emulator.  I think it just tries to run the package with `python -m`.

File: boston/README.md [hunk 5]
```diff
@@ -144,17 +142,36 @@ gcloud ml-engine local train --module-name=trainer.task \
...
+```
+
+* **Run in Google Cloud ML Engine:**
+
 ```
 gcloud ml-engine jobs submit training $JOB_NAME \
  --stream-logs \
  --runtime-version 1.10 \
- --job-dir=$JOB_DIR \
+ --job-dir=$GCS_JOB_DIR \
```
Comment (dizcology) [inline boston/README.md]: perhaps consider keeping it as `JOB_DIR`, to be consistent with the name of the flag.

File: census/README.md [hunk 6]
```diff
@@ -1,313 +1,322 @@
...
-Using census data which contains data a person's age, education, marital status, 
-and occupation (the features), we will try to predict whether or not the person earns 
-more than 50,000 dollars a year (the target label). We will train a logistic regression 
-model that, given an individual's information, outputs a number between 0 and 1—this can 
-be interpreted as the probability that the individual has an annual income of over 50,000 dollars.
-
-As a modeler and developer, think about how this data is used and the potential benefits and harm a model's predictions can cause. A model like this could reinforce societal biases and disparities. Is each feature relevant to the problem you want to solve or will it introduce bias? For more information, read about [ML fairness](https://developers.google.com/machine-learning/fairness-overview/).
-
-## Examples
+<h1>Overview</h1>
```
Comment (dizcology) [inline census/README.md]: use markdown.

File: census/README.md [hunk 7]
```diff
@@ -1,313 +1,322 @@
...
-EVAL_FILE=gs://cloud-samples-data/ml-engine/census/data/adult.test.csv
-```
+The best way to setup your GCP project is to use this section in this
+[tutorial](https://cloud.google.com/ml-engine/docs/tensorflow/getting-started-training-prediction#set-up-your-gcp-project).
 
-### \*Optional\* Use local training files.
+* **Environment setup:**
 
-Since TensorFlow - not the Cloud ML Engine - handles reading from GCS, you can run all commands below using these environment variables. However, if your network is slow or unreliable, you may want to download the files for local training.
+Virtual environments are strongly suggested, but not required. Installing this
```
Comment (dizcology) [inline census/README.md]: this is very general and not census-specific.  is there a better place for this?

File: census/README.md [hunk 8]
```diff
@@ -1,313 +1,322 @@
...
-gsutil cp $EVAL_FILE census_data/adult.test.csv
+* **Install dependencies**
 
-TRAIN_FILE=census_data/adult.data.csv
-EVAL_FILE=census_data/adult.test.csv
+Install the python dependencies. `pip install --upgrade -r requirements.txt`
+
+#
+
+* **How to satisfy Cloud ML Engine project structure requirements:**
```
Comment (dizcology) [inline census/README.md]: also this is not census-specific.  consider linking to the doc site instead of duplicating content here.

File: census/README.md [hunk 9]
```diff
@@ -1,313 +1,322 @@
...
+* **Google Cloud ML Engine**
+
+* **GCloud configuration:**
 
 ```
-export TRAIN_STEPS=1000
 DATE=`date '+%Y%m%d_%H%M%S'`
-export OUTPUT_DIR=census_$DATE
-rm -rf $OUTPUT_DIR
+export JOB_DIR=census_$DATE
```
Comment (dizcology) [inline census/README.md]: here it is `JOB_DIR` but it becomes `GCS_JOB_DIR` below.

File: census/README.md [hunk 10]
```diff
@@ -323,41 +332,47 @@ gsutil ls -r $GCS_JOB_DIR/export
...
+(Optional) Inspect the model binaries with the SavedModel CLI
 TensorFlow ships with a CLI that allows you to inspect the signature of exported binary files. To do this run:
 
 ```
 SIGNATURE_DEF_KEY=`saved_model_cli show --dir $MODEL_BINARIES --tag serve | grep "SignatureDef key:" | awk 'BEGIN{FS="\""}{print $2}' | head -1`
 saved_model_cli show --dir $MODEL_BINARIES --tag serve --signature_def $SIGNATURE_DEF_KEY
 ```
 
-### Run Online Predictions
+* **Run Online Predictions**
```
Comment (dizcology) [inline census/README.md]: this is no census-specific.  consider factoring it out as a generic guide to online predictions?  such a guide must already exist on the doc site; are we considering simply linking to the doc site?

File: census/README.md (hunk 9 above)
Comment (nnegrey) [reply census/README.md]: I liked that he switched between JOB_DIR for testing locally and then GCS_JOB_DIR when running on the cloud to help distinguish the two, otherwise they have to overwrite JOB_DIR and people might think they did that already and skip that step. But, I could also see the point for consistency to match the parameters.

Comment (gogasca) [reply census/README.md]: While I was going through the samples, (copy/paste) starting from running locally to running in Cloud ML, I missed updating this flag couple of times,  Now I'm directly forced to update it and implying where this data belongs.

File: census/README.md (hunk 8 above)
Comment (gogasca) [reply census/README.md]: Done

File: census/README.md (hunk 7 above)
Comment (gogasca) [reply census/README.md]: Agree, my goal was that this README.md to be self-contained with mininum back and forth between different docs. I can link the reference to CMLE site for virtualenv, but though would be useful to just tell Devs, run this command and move on to next step.

File: census/README.md (hunk 10 above)
Comment (gogasca) [reply census/README.md]: Agree, my goal was that this README.md to be self-contained with mininum back and forth between different docs. In this case this is Census and Iris specific, other samples inference like Flowers may be different.
